File: 品味Java的21種設計模式/dp_cpp/decorator/example6/GoodsSaleEbo.cpp
#include "GoodsSaleEbo.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace decorator
			{
				namespace example6
				{

					bool GoodsSaleEbo::sale(std::string user, std::string customer, SaleModel *saleModel)
					{
						puts(user+"保存了"+customer+"购买 "+saleModel+" 的销售数据");
						return true;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/decorator/example6/GoodsSaleEbo.java
package cn.javass.dp.decorator.example6;

public class GoodsSaleEbo implements GoodsSaleEbi {

    public boolean sale(String user, String customer, SaleModel saleModel) {
        System.out.println(user + "保存了" + customer + "购买 " + saleModel + " 的销售数据");
        return true;
    }
}
